src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,126 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Manages storage information for the program.
+ */
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads a list of tasks from the file path in the constructor object.
+     *
+     * @return List of tasks.
+     * @throws DukeException If the file could not be read.
+     */
+    public List<Task> load() throws DukeException {
+        List<Task> tasks = new ArrayList<>();
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] tokens = line.split("\\|");
+                if (tokens.length < 3) {
+                    throw new IoDukeException("The task file is corrupted");
+                }
+
+                Task task;
+                boolean done = Boolean.parseBoolean(tokens[1]);
+                String description = tokens[2];
+                SimpleDateFormat dateFormat = new SimpleDateFormat("d/M/yyyy HHmm");
+
+                switch (tokens[0]) {
+                case "T":
+                    task = new Todo(description, done);
+                    break;
+                case "D":
+                    task = new Deadline(description, dateFormat.parse(tokens[3]));
+                    break;
+                case "E":
+                    task = new Event(description, dateFormat.parse(tokens[3]));
+                    break;
+                default:
+                    throw new IoDukeException("Invalid task type found");
+                }
+
+                tasks.add(task);
+            }
+
+            reader.close();
+        } catch (FileNotFoundException e) {
+            throw new IoDukeException("Error opening task file for reading");
+        } catch (IOException e) {
+            throw new IoDukeException("Error closing file reader");
+        } catch (ParseException e) {
+            throw new IoDukeException("Error parsing date in task file");
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Saves a list of tasks to the file path in the constructor object.
+     *
+     * @throws DukeException If the file failed to save.
+     */
+    public void save(TaskList tasks) throws DukeException {
+        PrintWriter writer = null;
+        try {
+            Path path = Paths.get(filePath);
+
+            // Create directories if necessary
+            if (path.getParent() != null) {
+                Files.createDirectories(path.getParent().getFileName());
+            }
+
+            writer = new PrintWriter(new BufferedWriter(new FileWriter(filePath)));
```

Comment:
Might want to consider ObjectOutputStream to abstract writing and parsing objects
https://docs.oracle.com/javase/7/docs/api/java/io/ObjectOutputStream.html